// @patternfly/react-core/dist/umd/components/BackgroundImage/BackgroundImage.js
(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "prop-types", "react", "@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage", "@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_2x", "@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_sm", "@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_sm_2x", "@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_lg", "@patternfly/react-tokens/dist/js/c_background_image_Filter", "@patternfly/react-styles", "@patternfly/react-styles/css/components/BackgroundImage/background-image"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("prop-types"), require("react"), require("@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage"), require("@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_2x"), require("@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_sm"), require("@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_sm_2x"), require("@patternfly/react-tokens/dist/js/c_background_image_BackgroundImage_lg"), require("@patternfly/react-tokens/dist/js/c_background_image_Filter"), require("@patternfly/react-styles"), require("@patternfly/react-styles/css/components/BackgroundImage/background-image"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.propTypes, global.react, global.c_background_image_BackgroundImage, global.c_background_image_BackgroundImage_2x, global.c_background_image_BackgroundImage_sm, global.c_background_image_BackgroundImage_sm_2x, global.c_background_image_BackgroundImage_lg, global.c_background_image_Filter, global.reactStyles, global.backgroundImage);
    global.undefined = mod.exports;
  }
})(this, function (exports, _propTypes, _react, _c_background_image_BackgroundImage, _c_background_image_BackgroundImage_2x, _c_background_image_BackgroundImage_sm, _c_background_image_BackgroundImage_sm_2x, _c_background_image_BackgroundImage_lg, _c_background_image_Filter, _reactStyles, _backgroundImage) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.BackgroundImage = undefined;

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var React = _interopRequireWildcard(_react);

  var _c_background_image_BackgroundImage2 = _interopRequireDefault(_c_background_image_BackgroundImage);

  var _c_background_image_BackgroundImage_2x2 = _interopRequireDefault(_c_background_image_BackgroundImage_2x);

  var _c_background_image_BackgroundImage_sm2 = _interopRequireDefault(_c_background_image_BackgroundImage_sm);

  var _c_background_image_BackgroundImage_sm_2x2 = _interopRequireDefault(_c_background_image_BackgroundImage_sm_2x);

  var _c_background_image_BackgroundImage_lg2 = _interopRequireDefault(_c_background_image_BackgroundImage_lg);

  var _c_background_image_Filter2 = _interopRequireDefault(_c_background_image_Filter);

  var _backgroundImage2 = _interopRequireDefault(_backgroundImage);

  function _getRequireWildcardCache() {
    if (typeof WeakMap !== "function") return null;
    var cache = new WeakMap();

    _getRequireWildcardCache = function () {
      return cache;
    };

    return cache;
  }

  function _interopRequireWildcard(obj) {
    if (obj && obj.__esModule) {
      return obj;
    }

    var cache = _getRequireWildcardCache();

    if (cache && cache.has(obj)) {
      return cache.get(obj);
    }

    var newObj = {};

    if (obj != null) {
      var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;

      for (var key in obj) {
        if (Object.prototype.hasOwnProperty.call(obj, key)) {
          var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;

          if (desc && (desc.get || desc.set)) {
            Object.defineProperty(newObj, key, desc);
          } else {
            newObj[key] = obj[key];
          }
        }
      }
    }

    newObj.default = obj;

    if (cache) {
      cache.set(obj, newObj);
    }

    return newObj;
  }

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  function _extends() {
    _extends = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends.apply(this, arguments);
  }

  function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};

    var target = _objectWithoutPropertiesLoose(source, excluded);

    var key, i;

    if (Object.getOwnPropertySymbols) {
      var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

      for (i = 0; i < sourceSymbolKeys.length; i++) {
        key = sourceSymbolKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
        target[key] = source[key];
      }
    }

    return target;
  }

  function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;

    for (i = 0; i < sourceKeys.length; i++) {
      key = sourceKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      target[key] = source[key];
    }

    return target;
  }

  const defaultFilter = React.createElement("filter", null, React.createElement("feColorMatrix", {
    type: "matrix",
    values: "1 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 0 1 0"
  }), React.createElement("feComponentTransfer", {
    colorInterpolationFilters: "sRGB",
    result: "duotone"
  }, React.createElement("feFuncR", {
    type: "table",
    tableValues: "0.086274509803922 0.43921568627451"
  }), React.createElement("feFuncG", {
    type: "table",
    tableValues: "0.086274509803922 0.43921568627451"
  }), React.createElement("feFuncB", {
    type: "table",
    tableValues: "0.086274509803922 0.43921568627451"
  }), React.createElement("feFuncA", {
    type: "table",
    tableValues: "0 1"
  })));
  let filterCounter = 0;

  const BackgroundImage = exports.BackgroundImage = _ref => {
    let {
      className,
      src,
      filter = defaultFilter
    } = _ref,
        props = _objectWithoutProperties(_ref, ["className", "src", "filter"]);

    const getUrlValue = size => {
      if (typeof src === 'string') {
        return `url(${src})`;
      } else if (typeof src === 'object') {
        return `url(${src[size]})`;
      }

      return '';
    };

    const filterId = `patternfly-background-image-filter-overlay${filterCounter++}`;
    const style = {
      [_c_background_image_BackgroundImage2.default.name]: getUrlValue('xs'),
      [_c_background_image_BackgroundImage_2x2.default.name]: getUrlValue('xs2x'),
      [_c_background_image_BackgroundImage_sm2.default.name]: getUrlValue('sm'),
      [_c_background_image_BackgroundImage_sm_2x2.default.name]: getUrlValue('sm2x'),
      [_c_background_image_BackgroundImage_lg2.default.name]: getUrlValue('lg'),
      [_c_background_image_Filter2.default.name]: `url(#${filterId})`
    };
    return React.createElement("div", _extends({
      className: (0, _reactStyles.css)(_backgroundImage2.default.backgroundImage, className),
      style: style
    }, props), React.createElement("svg", {
      xmlns: "http://www.w3.org/2000/svg",
      className: "pf-c-background-image__filter",
      width: "0",
      height: "0"
    }, React.cloneElement(filter, {
      id: filterId
    })));
  };

  BackgroundImage.propTypes = {
    className: _propTypes2.default.string,
    filter: _propTypes2.default.element,
    src: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.shape({
      xs: _propTypes2.default.string.isRequired,
      xs2x: _propTypes2.default.string.isRequired,
      sm: _propTypes2.default.string.isRequired,
      sm2x: _propTypes2.default.string.isRequired,
      lg: _propTypes2.default.string.isRequired
    })]).isRequired
  };
});
